android/src/main/java/tech/bam/rnas/RNASModule.kt

```diff
@@ -2,9 +2,63 @@ package tech.bam.rnas
 
 import expo.modules.kotlin.modules.Module
 import expo.modules.kotlin.modules.ModuleDefinition
+import android.content.Context
+import android.view.inputmethod.InputMethodManager
+import android.provider.Settings
+import android.util.Log
+
 
 class RNASModule : Module() {
   override fun definition() = ModuleDefinition {
     Name("RNAS")
+
+    Function("showInputMethodPicker") {
+      val inputMethodManager = context.getSystemService(Context.INPUT_METHOD_SERVICE) as InputMethodManager
+      inputMethodManager.showInputMethodPicker()
+    }
+
+    Function("isCurrentKeyboardSafe") { customWhiteListedKeyboardPackages: Array<String>? ->
+      val keyboardID = getCurrentKeyboardID()
+      val isSafe = isKeyboardSafe(keyboardID, customWhiteListedKeyboardPackages)
+      // The Expo module system, especially for custom native modules, might not fully support synchronous
+      // returns depending on its version and the specific implementation.
+      return@Function isSafe
+    }
+  }
+
+  private val context
+  get() = requireNotNull(appContext.reactContext)
+
+  private val defaultWhiteListedKeyboardPackages = arrayOf("com.touchtype.swiftkey", "com.samsung.android", "com.google.android")
```

Comment:
let's use `allowListed` instead of `whiteListed` everywhere

Reply:
I used `defaultAllowedKeyboardPackagesList`

Reply:
why `allowed` and not `safe` ? since the function is called `isCurrentKeyboardSafe`